src/registrar/admin.py

```diff
@@ -353,6 +368,42 @@ class MyUserAdmin(BaseUserAdmin):
     # in autocomplete_fields for user
     ordering = ["first_name", "last_name", "email"]
 
+    def get_search_results(self, request, queryset, search_term):
```

Comment:
This is currently ONLY used for investigator. get_search_results, in the context of autocomplete, basically returns data from the actual table of this user object. 

This means that if we do a naive search and filter on investigator, the table will do that filter, too. We only want to do this from the context of the dropdowns though, so we have to be specific.

If, for example, in the future we want to filter "creator" - then this will be done here. Note that this ONLY applies to the User object. If we want to get more generic, we have to move this to the baseclass.